content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+ProxyPassMatch  ^/(?\!.well-known)  http://localhost:8080 nocanon
+----
+
+The _ProxyRequests Off_ prevents Apache from functioning as a forward
+proxy server (except for _ProxyPass_), it is advised to include it
+unless the server should function as a proxy.
+
+Both the `+nocanon+` option to `+ProxyPass+`, _and_
+`+AllowEncodedSlashes NoDecode+`, are required for certain Jenkins
+features to work.
+
+If you are running Apache on a Security-Enhanced Linux (SE-Linux)
+machine it is essential to make SE-Linux do the right thing by issuing
+as root
+
+[source]
+----
+setsebool -P httpd_can_network_connect true
+----
+
+If this is not issued Apache will not be allowed to forward proxy
+requests to Jenkins and only an error message will be displayed.
+
+Because Jenkins already compress its output, you can not use the normal
+proxy-html filter to modify urls:
+
+[source]
+----
+SetOutputFilter proxy-html
+----
+
+Instead you can use the following:
+
+[source]
+----
+SetOutputFilter INFLATE;proxy-html;DEFLATE
+ProxyHTMLURLMap http://your_server:8080/jenkins /jenkins
+----
+
+http://wiki.uniformserver.com/index.php/Reverse_Proxy_Server_2:_mod_proxy_html_2 +
+But since Jenkins seems to be well behaved it's even better to just not
+use SetOutputFilter and ProxyHTMLURLMap.
+
+If there are problems with Jenkins sometimes servicing random garbage
+pages, then the following may help:
+
+[source]
+----
+SetEnv proxy-nokeepalive 1
+----
+
+Some plug-ins determine URLs from client requests from Host header, so
+if you experience some problems with wrong URLs, you can try to switch
+on `+ProxyPreserveHost+` directive, which is switched off by default:
+
+[source]
+----
+ProxyPreserveHost On
+----
+
+===== mod_proxy with HTTPS
+
+If you'd like to run Jenkins with reverse proxy in HTTPS, one user
+reported that HTTPS needs to be terminated at Jenkins, not at the
+front-end Apache.
+See http://hudson.361315.n4.nabble.com/Hudson-behind-an-Apache-Reverse-Proxy-w-SSL-td370997.html[this
+e-mail thread] for more discussion.
+
+Note that you also may need to have access to your host via regular
+http, else the admin interface test will report a broken proxy setup.
+
+Alternatively, you can add an additional `+ProxyPassReverse+` directive
+to redirect non-SSL URLs generated by Jenkins to the SSL side.
+Assuming that your webserver is `+your.host.com+`, placing the following within
+the SSL virtual host definition will do the trick:
+
+[source]
+----
+ProxyRequests     Off
+ProxyPreserveHost On
+AllowEncodedSlashes NoDecode
+
+<Proxy http://localhost:8081/jenkins*>
+  Order deny,allow
+  Allow from all
+</Proxy>
+
+ProxyPass         /jenkins  http://localhost:8081/jenkins nocanon
+ProxyPassReverse  /jenkins  http://localhost:8081/jenkins
+ProxyPassReverse  /jenkins  http://your.host.com/jenkins
+----
+
+Yet another option is to rewrite the Location headers that contain
+non-ssl URL's generated by Jenkins.
+If you want to access Jenkins from https://www.example.com/jenkins,
```

Comment:
Escape the URL so that it is not converted into a clickable URL by asciidoc.
```suggestion
If you want to access Jenkins from \https://www.example.com/jenkins,
```